core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -823,24 +823,62 @@ protected function createFooterScripts($objLayout, $objPage = null)
 				$noSearch = true;
 			}
 
+			// schema.org
+			$isPreview = System::getContainer()->get('contao.security.token_checker')->isPreviewMode();
+
 			$meta = array
 			(
-				'@context' => 'https://contao.org/',
-				'@type' => 'PageMetaData',
-				'pageId' => (int) $objPage->id,
-				'language' => $objPage->language,
-				'title' => $objPage->pageTitle ?: $objPage->title,
-				'noSearch' => $noSearch,
-				'protected' => (bool) $objPage->protected,
-				'groups' => array_map('intval', array_filter((array) $objPage->groups)),
-				'fePreview' => System::getContainer()->get('contao.security.token_checker')->isPreviewMode()
+				'@context' => 'https://schema.org/',
+				'@type' => 'WebPage',
+				'name' => $objPage->pageTitle ?: $objPage->title,
+				'description' => $objPage->description,
+				'identifier' => (int) $objPage->id,
+				'inLanguage' => $objPage->language,
+				'version' => $isPreview ? 'preview' : 'current',
 			);
 
+			// Mark searchable
+			if (!$noSearch)
+			{
+				$meta['potentialAction']['@type'] = 'SearchAction';
+			}
+
+			// Mark protected
+			if ($objPage->protected)
+			{
+				$meta['conditionsOfAccess'] = 'Accessible only from logged-in accounts';
+			}
+
+			// Audience
+			$audience = array();
+
+			$groups = array_map('intval', array_filter((array) $objPage->groups));
+			if (!empty($groups))
+			{
+				$audience[] = array
+				(
+					'@type' => 'PeopleAudience',
+					'name' => 'groups',
+					'identifier' => $groups,
+				);
+			}
+
+			// Add member ID if logged in
 			$token = System::getContainer()->get('security.token_storage')->getToken();
 
 			if ($token !== null && $token->getUser() instanceof FrontendUser)
 			{
-				$meta['memberId'] = (int) $token->getUser()->id;
+				$audience[] = array
+				(
+					'@type' => 'PeopleAudience',
+					'name' => 'member',
+					'identifier' => (int) $token->getUser()->id,
+				);
+			}
```

Comment:
Why do we override the group permissions with user permissions? There can never be group permissions then, because there's always a user logged in when viewing that information? I know that has been the case before (and this PR is only to rewrite the existing), but maybe you can explain?